src/czbenchmarks/tasks/task.py

```diff
@@ -21,6 +29,294 @@ class TaskOutput(BaseModel):
     model_config = {"arbitrary_types_allowed": True}
 
 
+class TaskParameter(BaseModel):
+    """Schema for a single, discoverable parameter."""
+
+    type: str
+    default: Any = None
+    required: bool
+
+
+class TaskInfo(BaseModel):
+    """Schema for all discoverable information about a single benchmark task."""
+
+    name: str
+    display_name: str
+    description: str
+    task_params: Dict[str, TaskParameter]
+    baseline_params: Dict[str, TaskParameter]
+    metrics: List[str]
+
+
+class TaskRegistry:
+    """A registry that is populated automatically as Task subclasses are defined."""
+
+    def __init__(self):
+        self._registry: Dict[str, Type["Task"]] = {}
```

Comment:
If you `from __future__ import annotations` you shouldn't need to quote all the forward declarations for `Task`